src/windshaft/windshaft-map.js

```diff
@@ -71,10 +71,12 @@ var WindshaftMap = Backbone.Model.extend({
 
         // TODO: Revisit this (will layerIndex work for NamedMaps??)
         // Should we move it somewhere else?
-        _.each(layers, function (layer, layerIndex) {
+        _.each(options.layers, function (layer, layerIndex) {
           if (layer.get('type') === 'torque') {
             layer.set('meta', this.getLayerMeta(layerIndex));
             layer.set('urls', this.getTiles('torque'));
+          } else if (layer.get('type') === 'CartoDB' || layer.get('type') === 'mapnik') {
```

Comment:
In the context of CartoDB.js, CartoDB layers should have the `CartoDB` type so I don't understand why we are checking for `mapnik`...